### PR TITLE
fix(c++):  using C++17's nested namespaces 

### PR DESCRIPTION
**Pull request by ywh555hhh**
Initial modifications to feat(c++): Use C++17's nested namespaces #473 completed, involving content of anonymous namespace and content in result.hpp not changed yet

### Reason for this PR

to slove the issue #473

### What changes are included in this PR?

Most cpp folders need to use cpp17 nested loops
### Are these changes tested?

No local tests

### Are there any user-facing changes?

no

### CODE REVIEW

**Comment by ywh555hhh** [comment]
I seem to be ignoring some of the call logic for nested namespaces and I'll fix it myself

**Comment by acezen** [comment]
Hi @ywh555hhh, thanks for the work, the CI indicate that the code format check is failed. Please follow the https://github.com/apache/incubator-graphar/blob/87d5eeb6a7f86574328da26f08d950e3c5bc3de8/.github/workflows/ci.yml#L82-L83
to install clang-format 8.0 and fix the format with

```
make graphar-clformat
```

File: cpp/thirdparty/simple-uri-parser/uri_parser.h [hunk 1]
```diff
@@ -89,7 +89,7 @@ struct Uri {
         {}
 };
 
-}
```
Comment (acezen) [inline cpp/thirdparty/simple-uri-parser/uri_parser.h:92]: we better to keep the thirdparty code as their origin, not to modify them.

**Comment by ywh555hhh** [comment]
I encountered an error when trying to open a project in a container using vscode's remote container plug-in. How can I solve it? 

Error response from daemon: pull access denied for registry.cn-hongkong.aliyuncs.com/graphscope/graphar-dev, repository does not exist or may require 'docker login': denied: requested access to the resource is denied

**Comment by acezen** [comment]
> I encountered an error when trying to open a project in a container using vscode's remote container plug-in. How can I solve it?
> 
> Error response from daemon: pull access denied for registry.cn-hongkong.aliyuncs.com/graphscope/graphar-dev, repository does not exist or may require 'docker login': denied: requested access to the resource is denied

It seems that the image has been expired, I have re-pushed it, can you try again?

**Comment by ywh555hhh** [comment]
[info2.txt](https://github.com/apache/incubator-graphar/files/15400291/info2.txt)
I've tried it with both macos and linux and it still doesn't seem to work

**Comment by acezen** [comment]
> [info2.txt](https://github.com/apache/incubator-graphar/files/15400291/info2.txt) I've tried it with both macos and linux and it still doesn't seem to work

It seems that the image is a private repository, thanks for report the error. 
I have created an issue #491 to fix this.

**Comment by acezen** [comment]
> > [info2.txt](https://github.com/apache/incubator-graphar/files/15400291/info2.txt) I've tried it with both macos and linux and it still doesn't seem to work
> 
> It seems that the image is a private repository, thanks for report the error. I have created an issue #491 to fix this.

Hi, @ywh555hhh, the image has updated, could you rebase the main and try again, sorry for the obstruction.

**Comment by ywh555hhh** [comment]
I'm happy to contribute to the community, but I also want to know how to solve the formatting problem. If I can't use docker right now, it means I need to fix the problem on macOS

**Comment by acezen** [comment]
> https://github.com/muttleyxd/clang-tools-static-binaries/releases/download/master-22538c65/clang-format-8

the clang-format-8 is also available for macos:
```
curl -L https://github.com/muttleyxd/clang-tools-static-binaries/releases/download/master-22538c65/clang-format-8_macosx-amd64 --output clang-format
```

File: cpp/include/graphar/util/filesystem.h [hunk 2]
```diff
@@ -35,16 +35,19 @@
 namespace arrow {
 class Buffer;
 class Table;
-namespace fs {
+}// namespace arrow
```
Comment (acezen) [inline cpp/include/graphar/util/filesystem.h:38]: the correct format is 
```suggestion
} // namespace arrow
```

File: cpp/include/graphar/util/filesystem.h [hunk 3]
```diff
@@ -35,16 +35,19 @@
...
+
+namespace arrow::fs {
 class FileSystem;
-}
-namespace io {
+}// namespace arrow::fs
```
Comment (acezen) [inline cpp/include/graphar/util/filesystem.h:42]: Ditto

File: cpp/include/graphar/util/filesystem.h [hunk 4]
```diff
@@ -35,16 +35,19 @@
...
+
+namespace arrow::io {
 class RandomAccessFile;
-}
-namespace dataset {
+}// namespace arrow::io
```
Comment (acezen) [inline cpp/include/graphar/util/filesystem.h:46]: Ditto

File: cpp/include/graphar/fwd.h [hunk 5]
```diff
@@ -201,12 +201,10 @@ const std::shared_ptr<DataType>& timestamp();
  * @param value_type value type of the list
  */
 std::shared_ptr<DataType> list(const std::shared_ptr<DataType>& value_type);
-
-namespace util {
+}  // namespace graphar
```
Comment (acezen) [inline cpp/include/graphar/fwd.h:204]: here should have an empty line?

File: cpp/include/graphar/util/result.h [hunk 6]
```diff
@@ -117,6 +115,4 @@ inline Status GenericToStatus(Result<T>&& res) {
   return std::move(res).status();
 }
 
-}  // namespace internal
-
-}  // namespace graphar
+}  // namespace graphar::internal
```
Comment (acezen) [inline cpp/include/graphar/util/result.h:118]: here should be an empty line in the last of file

File: cpp/include/graphar/writer/edges_builder.h [hunk 7]
```diff
@@ -426,5 +424,4 @@ class EdgesBuilder {
   ValidateLevel validate_level_;
 };
 
-}  // namespace builder
-}  // namespace graphar
+}  // namespace graphar::builder
```
Comment (acezen) [inline cpp/include/graphar/writer/edges_builder.h:427]: Add an empty line in the tail of source file

File: cpp/include/graphar/writer/vertices_builder.h [hunk 8]
```diff
@@ -347,5 +346,4 @@ class VerticesBuilder {
   ValidateLevel validate_level_;
 };
 
-}  // namespace builder
-}  // namespace graphar
+}  // namespace graphar::builder
```
Comment (acezen) [inline cpp/include/graphar/writer/vertices_builder.h:349]: Ditto

File: cpp/thirdparty/simple-uri-parser/uri_parser.h [hunk 9]
```diff
@@ -89,7 +89,7 @@ struct Uri {
...
+}
```
Comment (acezen) [inline cpp/thirdparty/simple-uri-parser/uri_parser.h:92]: please trim the space.

File: cpp/src/edges_builder.cc [hunk 10]
```diff
@@ -349,5 +348,4 @@ Result<std::shared_ptr<arrow::Table>> EdgesBuilder::getOffsetTable(
   return arrow::Table::Make(schema, arrays);
 }
 
-}  // namespace builder
-}  // namespace graphar
+}  // namespace graphar::builder
```
Comment (acezen) [inline cpp/src/edges_builder.cc:351]: Ditto

File: cpp/src/filesystem.cc [hunk 11]
```diff
@@ -311,4 +310,4 @@ template Result<IdType> FileSystem::ReadFileToValue<IdType>(
 template Status FileSystem::WriteValueToFile<IdType>(const IdType&,
                                                      const std::string&) const
     noexcept;
-}  // namespace graphar
+}  // namespace graphar
```
Comment (acezen) [inline cpp/src/filesystem.cc:313]: Ditto

File: cpp/src/reader_util.cc [hunk 12]
```diff
@@ -206,6 +204,4 @@ Result<IdType> GetEdgeNum(const std::string& prefix,
   return edge_num;
 }
 
-}  // namespace util
-
-}  // namespace graphar
+}  // namespace graphar::util
```
Comment (acezen) [inline cpp/src/reader_util.cc:207]: Ditto

File: cpp/src/util.cc [hunk 13]
```diff
@@ -99,5 +97,4 @@ std::string ValueGetter<std::string>::Value(const void* data, int64_t offset) {
       reinterpret_cast<const arrow::LargeStringArray*>(data)->GetView(offset));
 }
 
-}  // namespace util
-}  // namespace graphar
+}  // namespace graphar::util
```
Comment (acezen) [inline cpp/src/util.cc:100]: Ditto

File: cpp/src/vertices_builder.cc [hunk 14]
```diff
@@ -232,5 +231,4 @@ Result<std::shared_ptr<arrow::Table>> VerticesBuilder::convertToTable() {
   return arrow::Table::Make(schema, arrays);
 }
 
-}  // namespace builder
-}  // namespace graphar
+}  // namespace graphar::builder
```
Comment (acezen) [inline cpp/src/vertices_builder.cc:234]: Ditto

**Comment by acezen** [comment]
It's better not to delete the empty line in the end of the source file. FYI
https://stackoverflow.com/questions/2287967/why-is-it-recommended-to-have-empty-line-in-the-end-of-a-source-file

**Review by acezen — COMMENTED** [review]
Seems that you have change the permission of `cpp/misc/cpplint.py`, is there a reason to change this?

**Comment by ywh555hhh** [comment]
> Seems that you have changed the permission of `cpp/misc/cpplint.py`, is there a reason to change this?
I had no intention of doing that. Could this have been some kind of operational error that I wasn't aware of. Sorry

**Comment by acezen** [comment]
> > Seems that you have changed the permission of `cpp/misc/cpplint.py`, is there a reason to change this?
> 
> I had no intention of doing that. Could this have been some kind of operational error that I wasn't aware of. Sorry

It would make the CI failed. You can revert it to the original commit with git
https://github.com/apache/incubator-graphar/actions/runs/9190636917/job/25275644225?pr=489

**Comment by ywh555hhh** [comment]
> > > Seems that you have changed the permission of `cpp/misc/cpplint.py`, is there a reason to change this?
> > 
> > 
> > I had no intention of doing that. Could this have been some kind of operational error that I wasn't aware of. Sorry
> 
> It would make the CI failed. see https://github.com/apache/incubator-graphar/actions/runs/9190636917/job/25275644225?pr=489
> 
> You can revert it to the original commit with git https://stackoverflow.com/questions/215718/how-can-i-reset-or-revert-a-file-to-a-specific-revision

Thanks for the Pointers. I really appreciate you taking the time to mentor me

File: cpp/src/filesystem.cc (hunk 11 above)
Comment (acezen) [reply cpp/src/filesystem.cc:313]: better to add empty line to the end of this file.

**Comment by acezen** [comment]
> > > > Seems that you have changed the permission of `cpp/misc/cpplint.py`, is there a reason to change this?
> > > 
> > > 
> > > I had no intention of doing that. Could this have been some kind of operational error that I wasn't aware of. Sorry
> > 
> > 
> > It would make the CI failed. see https://github.com/apache/incubator-graphar/actions/runs/9190636917/job/25275644225?pr=489
> > You can revert it to the original commit with git https://stackoverflow.com/questions/215718/how-can-i-reset-or-revert-a-file-to-a-specific-revision
> 
> Thanks for the Pointers. I really appreciate you taking the time to mentor me

You are welcome. We are pleased to have your contribution.

File: cpp/include/graphar/util/filesystem.h [hunk 15]
```diff
@@ -154,4 +154,3 @@ Result<std::shared_ptr<FileSystem>> FileSystemFromUriOrPath(
     const std::string& uri, std::string* out_path = nullptr);
 
 }  // namespace graphar
-
```
Comment (acezen) [inline cpp/include/graphar/util/filesystem.h]: why delete the empty line of the file?

File: cpp/include/graphar/util/reader_util.h [hunk 16]
```diff
@@ -72,4 +72,3 @@ Result<IdType> GetEdgeNum(const std::string& prefix,
                           IdType vertex_chunk_index) noexcept;
 
 }  // namespace graphar::util
-
```
Comment (acezen) [inline cpp/include/graphar/util/reader_util.h]: why delete the empty line of the file?

File: cpp/include/graphar/util/filesystem.h (hunk 15 above)
Comment (ywh555hhh) [reply cpp/include/graphar/util/filesystem.h]: I left a blank line in my editor, and I felt a little weird coming across this situation

Comment (acezen) [reply cpp/include/graphar/util/filesystem.h]: Maybe your editor has configured to trim the empty line automatically? You can check about this

**Comment by ywh555hhh** [comment]
The version of clang-format I used was wrong and I will try to install the required version myself

**Comment by ywh555hhh** [comment]
```bash
yiweihan@yiweihandeMacBook-Air cpp %  clang-format --version
clang-format version 8.0.1 (tags/RELEASE_801/final)
yiweihan@yiweihandeMacBook-Air cpp % make clformat          
make: *** No rule to make target `clformat'.  Stop.
yiweihan@yiweihandeMacBook-Air cpp % 

```
I have successfully installed clang-format but I can't run the clformat command. What is the problem? And the cpp folder that I noticed contains the.clang-format file, which says the format standard

**Comment by acezen** [comment]
> ```shell
> yiweihan@yiweihandeMacBook-Air cpp %  clang-format --version
> clang-format version 8.0.1 (tags/RELEASE_801/final)
> yiweihan@yiweihandeMacBook-Air cpp % make clformat          
> make: *** No rule to make target `clformat'.  Stop.
> yiweihan@yiweihandeMacBook-Air cpp % 
> ```
> 
> I have successfully installed clang-format but I can't run the clformat command. What is the problem? And the cpp folder that I noticed contains the.clang-format file, which says the format standard

the command 
```
make graphar-clformat
```
is run in the build directory if you create a build dir

File: cpp/include/graphar/util/reader_util.h [hunk 17]
```diff
@@ -25,10 +25,7 @@
 #include <vector>
 
 #include "graphar/fwd.h"
-
-namespace graphar {
-
-namespace util {
+namespace graphar::util {
```
Comment (acezen) [inline cpp/include/graphar/util/reader_util.h:28]: here should have an empty line between header include and the implementation code.

File: cpp/include/graphar/util/status.h [hunk 18]
```diff
@@ -60,10 +60,7 @@
     ::graphar::Status __s = ::graphar::internal::GenericToStatus(status); \
     GAR_RAISE_ERROR_IF_(!__s.ok(), __s, GAR_STRINGIFY(status));           \
   } while (false)
-
-namespace graphar {
-
-namespace util {
+namespace graphar::util {
```
Comment (acezen) [inline cpp/include/graphar/util/status.h:63]: add an empty line before namespace for code clarity

File: cpp/include/graphar/writer/edges_builder.h [hunk 19]
```diff
@@ -35,9 +35,7 @@
 namespace arrow {
 class Array;
 }
-
-namespace graphar {
-namespace builder {
+namespace graphar::builder {
```
Comment (acezen) [inline cpp/include/graphar/writer/edges_builder.h:38]: add an empty line before namespace for code clarity

File: cpp/src/filesystem.cc [hunk 20]
```diff
@@ -30,10 +30,7 @@
 #include "graphar/fwd.h"
 #include "graphar/util/expression.h"
 #include "graphar/util/filesystem.h"
-
-namespace graphar {
-namespace ds = arrow::dataset;
-namespace detail {
+namespace graphar::detail {
```
Comment (acezen) [inline cpp/src/filesystem.cc:33]: here need empty line between header include and implementation

**Comment by ywh555hhh** [comment]
> > ```shell
> > yiweihan@yiweihandeMacBook-Air cpp %  clang-format --version
> > clang-format version 8.0.1 (tags/RELEASE_801/final)
> > yiweihan@yiweihandeMacBook-Air cpp % make clformat          
> > make: *** No rule to make target `clformat'.  Stop.
> > yiweihan@yiweihandeMacBook-Air cpp % 
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I have successfully installed clang-format but I can't run the clformat command. What is the problem? And the cpp folder that I noticed contains the.clang-format file, which says the format standard
> 
> the command
> 
> ```
> make graphar-clformat
> ```
> 
> is run in the build directory if you create a build dir


```
yiweihan@yiweihandeMacBook-Air cpp %  clang-format --version
clang-format version 8.0.1 (tags/RELEASE_801/final)
yiweihan@yiweihandeMacBook-Air cpp % make clformat          
make: *** No rule to make target `clformat'.  Stop.
yiweihan@yiweihandeMacBook-Air cpp % make graphar-clformat
make: *** No rule to make target `graphar-clformat'.  Stop.
yiweihan@yiweihandeMacBook-Air cpp % cd ..
yiweihan@yiweihandeMacBook-Air incubator-graphar % mkdir build
yiweihan@yiweihandeMacBook-Air incubator-graphar % cd build
yiweihan@yiweihandeMacBook-Air build % make graphar-clformat
make: *** No rule to make target `graphar-clformat'.  Stop.
yiweihan@yiweihandeMacBook-Air build % 
```
Did I misunderstand something? I don't even seem to be looking for the Makefile

File: cpp/src/filesystem.cc [hunk 21]
```diff
@@ -79,7 +76,9 @@ static Status CastToLargeOffsetArray(
   GAR_RETURN_ON_ARROW_ERROR_AND_ASSIGN(out, arrow::ChunkedArray::Make(chunks));
   return Status::OK();
 }
-}  // namespace detail
+}  // namespace graphar::detail
```
Comment (acezen) [inline cpp/src/filesystem.cc:79]: Add empty line here for code clarify

**Review by acezen — COMMENTED** [review]
Overall LGTM, and there have some nits to be fixed

**Comment by acezen** [comment]
> > > ```shell
> > > yiweihan@yiweihandeMacBook-Air cpp %  clang-format --version
> > > clang-format version 8.0.1 (tags/RELEASE_801/final)
> > > yiweihan@yiweihandeMacBook-Air cpp % make clformat          
> > > make: *** No rule to make target `clformat'.  Stop.
> > > yiweihan@yiweihandeMacBook-Air cpp % 
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > I have successfully installed clang-format but I can't run the clformat command. What is the problem? And the cpp folder that I noticed contains the.clang-format file, which says the format standard
> > 
> > 
> > the command
> > ```
> > make graphar-clformat
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > is run in the build directory if you create a build dir
> 
> ```
> yiweihan@yiweihandeMacBook-Air cpp %  clang-format --version
> clang-format version 8.0.1 (tags/RELEASE_801/final)
> yiweihan@yiweihandeMacBook-Air cpp % make clformat          
> make: *** No rule to make target `clformat'.  Stop.
> yiweihan@yiweihandeMacBook-Air cpp % make graphar-clformat
> make: *** No rule to make target `graphar-clformat'.  Stop.
> yiweihan@yiweihandeMacBook-Air cpp % cd ..
> yiweihan@yiweihandeMacBook-Air incubator-graphar % mkdir build
> yiweihan@yiweihandeMacBook-Air incubator-graphar % cd build
> yiweihan@yiweihandeMacBook-Air build % make graphar-clformat
> make: *** No rule to make target `graphar-clformat'.  Stop.
> yiweihan@yiweihandeMacBook-Air build % 
> ```
> 
> Did I misunderstand something? I don't even seem to be looking for the Makefile

you need to run `cmake ..` before `make`. It's the same with building process

**Comment by ywh555hhh** [comment]
> > > > ```shell
> > > > yiweihan@yiweihandeMacBook-Air cpp %  clang-format --version
> > > > clang-format version 8.0.1 (tags/RELEASE_801/final)
> > > > yiweihan@yiweihandeMacBook-Air cpp % make clformat          
> > > > make: *** No rule to make target `clformat'.  Stop.
> > > > yiweihan@yiweihandeMacBook-Air cpp % 
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > I have successfully installed clang-format but I can't run the clformat command. What is the problem? And the cpp folder that I noticed contains the.clang-format file, which says the format standard
> > > 
> > > 
> > > the command
> > > ```
> > > make graphar-clformat
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > is run in the build directory if you create a build dir
> > 
> > 
> > ```
> > yiweihan@yiweihandeMacBook-Air cpp %  clang-format --version
> > clang-format version 8.0.1 (tags/RELEASE_801/final)
> > yiweihan@yiweihandeMacBook-Air cpp % make clformat          
> > make: *** No rule to make target `clformat'.  Stop.
> > yiweihan@yiweihandeMacBook-Air cpp % make graphar-clformat
> > make: *** No rule to make target `graphar-clformat'.  Stop.
> > yiweihan@yiweihandeMacBook-Air cpp % cd ..
> > yiweihan@yiweihandeMacBook-Air incubator-graphar % mkdir build
> > yiweihan@yiweihandeMacBook-Air incubator-graphar % cd build
> > yiweihan@yiweihandeMacBook-Air build % make graphar-clformat
> > make: *** No rule to make target `graphar-clformat'.  Stop.
> > yiweihan@yiweihandeMacBook-Air build % 
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Did I misunderstand something? I don't even seem to be looking for the Makefile
> 
> you need to run `cmake ..` before `make`. It's the same with building process

tks I've tried it. Can this be merged now?

**Comment by ywh555hhh** [comment]
> Overall LGTM, and there have some nits to be fixed

ok👌🏻

**Review by acezen — APPROVED** [review]

**Pull request merged**